### PR TITLE
[Python] Mark variable annotation with a `meta.` block

### PR DESCRIPTION
**Pull request by kaste**
Use ~`meta.variable.parameters.annotation.python`~ `meta.variable.annotation.python` to mark annotation blocks when assigning.

This closely follows the `meta.function.parameters.annotation.python` we have in the function signatures.

### CODE REVIEW

**Comment by kaste** [comment]
E.g. the function signature could (always) be styled as  

image

After this patch, simple assignements can follow 

image

This is just a clear-as-possible example.  Just setting a background will keep the normal syntax highlighting of the tokens in-tact. 

Note.  I wanted to add tests for 

```
foo: \
    bar = zoo

zoo: \
    loo \
    = too
```
 which works

image

but as soon as I start the test line with an `#` the pattern pops (because it must pop for python comments; I added that.)

In other words it makes this

image

and `bar = zoo` are not marked anymore.

File: Python/Python.sublime-syntax [hunk 1]
```diff
@@ -1016,8 +1016,17 @@ contexts:
 
   assignments:
     - include: illegal-assignment-expression
-    - match: ':'
-      scope: punctuation.separator.annotation.variable.python
+    - match: '(?=:)'
+      set:
```
Comment (deathaxe) [inline Python/Python.sublime-syntax:1020]: 1. This would need to be a `push` as `assignments` is included in `statements` which is part of `main`.

2. Why adding the lookahead?

File: Python/Python.sublime-syntax [hunk 2]
```diff
@@ -1016,8 +1016,17 @@ contexts:
...
+        - match: ':'
+          scope: punctuation.separator.annotation.variable.python
+          set:
+            - meta_scope: meta.variable.parameters.annotation.python
```
Comment (deathaxe) [inline Python/Python.sublime-syntax:1024]: If we expect annotations to be primarily used for type-hints existing `meta.function.parameters.annotation` is not very suiteable anymore. The concept of `parameters` in assignments feels even less appropriate, IMHO.

`meta.annotation` would generally be a good scope, but it is unfortunatelly used/reserved for member attributes/modifiers in various syntaxes

example:

```py
@annotation
def func():
    pass
```

I have not final proposal for a suitable scope, but `meta.type.php` was introduced for type annotations in PHP recently. Maybe we want to use something like that for all kinds of type hints - also in function parameter lists and return types.

File: Python/Python.sublime-syntax [hunk 3]
```diff
@@ -1016,8 +1016,17 @@ contexts:
...
+            - match: '(?==|$|#)'
+              pop: 1
+            - include: expression-in-a-group
```
Comment (deathaxe) [inline Python/Python.sublime-syntax:1027]: This is a good starting point, but I guess we want to use the same context used by function parameters to also fix #3486. I've starded work for such a fix at https://github.com/deathaxe/sublime-packages/tree/pr/python/add-type-hints some weeks ago, but got stuck due to work on Package Control and real life.

**Comment by deathaxe** [comment]
In case of testing line continuation - it requires several tests to test such constructs. First test testing line 1, 2nd test testing line 2 (with content of 1 included) and so forth

```
vari = expr \
#           ^ punctuation


vari = expr \
  expr1 \
#       ^ punctuation

vari = expr \
  expr1 \
  expr2 \
#       ^ punctuation

...
```

File: Python/Python.sublime-syntax (hunk 1 above)
Comment (kaste) [reply Python/Python.sublime-syntax:1020]: Both because it is copied from `function-parameters: `.  I push a change so we can compare.  Don't know if I get this to work actually.

Comment (kaste) [reply Python/Python.sublime-syntax]: Ok, I pushed a commit and it checks green.

File: Python/Python.sublime-syntax (hunk 3 above)
Comment (kaste) [reply Python/Python.sublime-syntax]: It was meant as a babystep anyway.  I do not like the "invalid" markers actually; they tend to produce problems in the distant future; and that is a case here probably.

File: Python/Python.sublime-syntax (hunk 2 above)
Comment (kaste) [reply Python/Python.sublime-syntax]: Yeah `meta.variables.parameters` is non-sense.  Simply `meta.variable.annotation.python` is a better fit here.

We should not restrict to type-hints though.  These still can be used for meta-programming.

**Review by deathaxe — APPROVED** [review]

File: Python/Python.sublime-syntax [hunk 4]
```diff
@@ -1018,6 +1018,11 @@ contexts:
     - include: illegal-assignment-expression
     - match: ':'
       scope: punctuation.separator.annotation.variable.python
+      push:
+        - meta_scope: meta.variable.annotation.python
+        - match: '(?==|$|#)'
```
Comment (deathaxe) [inline Python/Python.sublime-syntax:1023]: We should probably include `,` and `;` to make sure to pop annotations in lists or multiple inline expressions

```suggestion
        - match: $|(?=[,;=#])
```

```
var1: Type1, var2: Type2;print("foo")
```

Comment (kaste) [reply Python/Python.sublime-syntax:1023]: The `,` is a syntax error here.  

image

The `;` is allowed though and should pop of course.

Comment (kaste) [reply Python/Python.sublime-syntax]: Done, still green

File: Python/Python.sublime-syntax [hunk 5]
```diff
@@ -1018,6 +1018,11 @@ contexts:
     - include: illegal-assignment-expression
     - match: ':'
       scope: punctuation.separator.annotation.variable.python
+      push:
+        - meta_scope: meta.variable.annotation.python
+        - match: (?=$|=|#|;)
+          pop: 1
+        - include: expression-in-a-group
```
Comment (FichteFoll) [inline Python/Python.sublime-syntax:1025]: Could you introduce a named context for this? We generally try to avoid inline contexts since syntax definitions can inherit/extend others since it allows more finely tuning the exact contexts you need.

**Review by FichteFoll — CHANGES_REQUESTED** [review]

File: Python/Python.sublime-syntax (hunk 5 above)
Comment (kaste) [reply Python/Python.sublime-syntax]: Done

**Review by FichteFoll — APPROVED** [review]

**Review by deathaxe — APPROVED** [review]
